app/controllers/investigations/alerts_controller.rb

```diff
@@ -1,55 +1,59 @@
 class Investigations::AlertsController < ApplicationController
-  include Wicked::Wizard
   include Pundit
   include ActionView::Helpers::NumberHelper
 
-  steps :about_alerts, :compose, :preview
-
-  before_action :set_investigation
-  before_action :set_alert, only: %i[show update], if: -> { %i[compose preview].include? step }
-  before_action :store_alert, only: :update, if: -> { step == :compose }
-  before_action :set_user_count, only: %i[show update create], if: -> { step == :preview }
-  before_action :get_preview, only: :show, if: -> { step == :preview }
+  def about
+    set_and_authorize_investigation
+    @investigation = @investigation.decorate
+  end
 
   def new
-    clear_session
-    redirect_to wizard_path(steps.first)
-  end
+    set_and_authorize_investigation
+    authorize @investigation, :investigation_restricted?
 
-  def show
-    render_wizard
+    @alert_form = AlertForm.new(alert_request_params.merge(investigation_url: investigation_url(@investigation)))
+    @investigation = @investigation.decorate
   end
 
-  def update
-    if @alert.valid?
-      return create if step == steps.last
+  def preview
+    set_and_authorize_investigation
+    authorize @investigation, :investigation_restricted?
 
-      redirect_to next_wizard_path
-    else
-      render_wizard
-    end
+    @alert_form = AlertForm.new(alert_request_params.merge(investigation_url: investigation_url(@investigation)))
+    @investigation = @investigation.decorate
+
+    return render :new unless @alert_form.valid?
+
+    set_user_count
+    get_preview
   end
 
   def create
-    @alert.save!
-    redirect_to investigation_path(@investigation), flash: { success: "Email alert sent to #{@user_count} users" }
-  end
+    set_and_authorize_investigation
+    authorize @investigation, :investigation_restricted?
 
-private
+    @alert_form = AlertForm.new(alert_request_params)
 
-  def clear_session
-    session.delete(:alert)
-  end
+    if @alert_form.valid?
+      AddAlert.call!(
+        @alert_form.attributes.merge({
+          investigation: @investigation,
+          user: current_user
+        })
+      )
 
-  def set_investigation
-    @investigation = Investigation.find_by!(pretty_id: params[:investigation_pretty_id])
-    authorize_investigation
+      @investigation = @investigation.decorate
+      redirect_to investigation_path(@investigation), flash: { success: "Email alert sent to users" }
+    else
+      render :new
```

Comment:
```suggestion
      @investigation = @investigation.decorate
      render :new
```
I think we need to decorate here.